code/modules/virus2/centrifuge.dm

```diff
@@ -563,7 +563,8 @@
 /obj/machinery/disease2/centrifuge/breakdown()
 	for (var/i = 1 to vial_data.len)
 		var/isolation_centrifuge_vial/vial_datum = vial_data[i]
-		vial_datum.vial.forceMove(loc)
+		if(vial_datum?.vial)
+			vial_datum.vial.forceMove(loc)
```

Comment:
Can't check this without testing but can't you just replace this entire if statement with:
`vial_datum?.vial.forceMove(loc)`

Reply:
The if statement in this PR is the equivalent of ``if(vial_datum && vial_datum.vial)``, this proposed change would need to be written as ``vial_datum?.vial?.forceMove(loc)`` to be correct. It's the same speed either way. I'd recommend writing it as a clear if statement if only for future coder clarity.

@SECBATON-GRIFFON feel free to change or mark as resolved.

Reply:
wait so your change would work?

Reply:
``if(vial_datum && vial_datum.vial)`` works because if ``vial_datum`` doesn't exist, byond doesn't even attempt to resolve the second part of the && since it now knows the whole thing will be false. Saves processing time and also dodges the runtime. Idential to how ``if(vial_datum?.vial)`` works, or how the single line ``vial_datum?.vial?.forceMove(loc)`` works.

https://www.byond.com/docs/ref/#/operator/%3f%2e

Reply:
done it now